Userland/Libraries/LibGfx/ImageFormats/FLICLoader.cpp

```diff
@@ -175,42 +177,64 @@ static ErrorOr<void> decode_frame(FLICLoadingContext& context, size_t frame_inde
             dbgln_if(FLIC_DEBUG, "Decoding delta_fli {}x{} buffer_size={}", context.width, context.height, stream->size());
             auto lines_to_skip = TRY(stream->read_value<LittleEndian<u16>>());
             auto number_of_lines = TRY(stream->read_value<LittleEndian<u16>>());
-            auto& previous_frame = context.frames.at(frame_index - 1);
 
             for (size_t line = lines_to_skip; line < lines_to_skip + number_of_lines; line++) {
                 size_t x = 0;
                 auto packets = TRY(stream->read_value<u8>());
 
                 for (u8 packet = 0; packet < packets; packet++) {
                     auto col_skip_count = TRY(stream->read_value<u8>());
-                    auto type = TRY(stream->read_value<i8>());
                     x += col_skip_count;
 
-                    // copy next pixels
+                    auto type = TRY(stream->read_value<i8>());
+
                     if (type > 0) {
-                        auto dest_x = x;
                         for (i8 n = 0; n < type; n++) {
                             auto color = TRY(stream->read_value<u8>());
-                            context.frame_buffer->set_pixel(dest_x++, line, previous_frame->color_map[color]);
+                            context.frame_buffer->set_pixel(x++, line, current_frame->color_map[color]);
                         }
                     } else if (type < 0) {
                         type = -type;
                         // repeat next pixels
                         auto color = TRY(stream->read_value<u8>());
-                        auto dest_x = x;
                         for (i8 n = 0; n < type; n++) {
-                            context.frame_buffer->set_pixel(dest_x++, line, previous_frame->color_map[color]);
+                            context.frame_buffer->set_pixel(x++, line, current_frame->color_map[color]);
                         }
                     }
+                }
+            }
+            break;
```

Comment:
This look like fixes for the previous commit, if so, please add them at the correct place.